boa/src/object/internal_methods.rs

```diff
@@ -390,6 +392,70 @@ impl GcObject {
         }
     }
 
+    /// Gets own property of 'Object'
+    ///
+    #[inline]
+    pub fn get_own_property(&self, key: &PropertyKey) -> Option<PropertyDescriptor> {
+        let _timer = BoaProfiler::global().start_event("Object::get_own_property", "object");
+
+        let object = self.borrow();
+        match object.data {
+            ObjectData::String(_) => self.string_exotic_get_own_property(key),
+            _ => self.ordinary_get_own_property(key),
+        }
+    }
+
+    /// StringGetOwnProperty abstract operation
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-stringgetownproperty
+    #[inline]
+    pub fn string_get_own_property(&self, key: &PropertyKey) -> Option<PropertyDescriptor> {
+        let object = self.borrow();
+
+        match key {
+            PropertyKey::Index(index) => {
+                let string = object.as_string().unwrap();
+                let pos = *index as usize;
+
+                if pos >= string.len() {
+                    return None;
+                }
+
+                let char = if let Some(utf16_val) = string.encode_utf16().nth(pos) {
+                    Value::from(from_u32(utf16_val as u32).unwrap())
```

Comment:
I think we just need to specify that it is `char`'s associated function, there is no need to have ` use std::char::from_u32;`, which makes is confusing what `from_u32` this is.
```suggestion
                    Value::from(char::from_u32(utf16_val as u32).unwrap())
```

Reply:
Can we do `u32::from(utf16_val)` here?, and use `expect()` instead of `unwrap()`.